### PR TITLE
[DG] Promoting query-param-content-type to the toolbox

### PR DESCRIPTION
**Pull request by danielglauser**
It's time we stopped copying this interceptor around and started
sharing the code.

### CODE REVIEW

File: src/pedestal_toolbox/params.clj [hunk 1]
```diff
@@ -68,6 +68,21 @@
             (assoc ctx :response (response/bad-request e)))))})
     {:schema schema}))
 
+(def query-param-content-type
+  "An enter interceptor that fakes an Accept header so that later
```
Comment (cap10morgan) [inline src/pedestal_toolbox/params.clj]: Should be "before interceptor." Yes the terminology is annoyingly inconsistent with the keywords you use to define them. But that's what they call them.

File: src/pedestal_toolbox/params.clj [hunk 2]
```diff
@@ -68,6 +68,21 @@
...
+  interceptors can handle the Accept header normally. This is used
+  because it's literally impossible to make a clickable link in a
+  browser that sets the Accept header in the normal way.
+
+  To use, add an content-type=application/csv to the URL query string.
```
Comment (cap10morgan) [inline src/pedestal_toolbox/params.clj]: s/an/a/ #pedantic

File: src/pedestal_toolbox/params.clj [hunk 3]
```diff
@@ -68,6 +68,21 @@
...
+  Expects the query params to have already been kewordized by kewordize-params"
```
Comment (cap10morgan) [inline src/pedestal_toolbox/params.clj]: Missing the "y" in "keywordized" (both times).

File: src/pedestal_toolbox/params.clj [hunk 4]
```diff
@@ -68,6 +68,21 @@
...
+  (interceptor
+   {:enter
+    (fn [ctx]
+      (if-let [content-type (get-in ctx [:request :params :content-type])]
+        (assoc-in ctx [:request :headers "accept"] content-type)
+        ctx))}))
```
Comment (cap10morgan) [inline src/pedestal_toolbox/params.clj]: I don't like how we're confusing content-type and accept in this code. "content-type" is for specifying the encoding of data you're sending to the other side (server to client or client to server), "accept" is for requesting the format you'd like your response encoded in.

Can we change the query param to "accept"?

**Comment by danielglauser** [comment]
Backatcha @cap10morgan

**Comment by cap10morgan** [comment]
Sorry this fell off my radar for so long. I think the only other thing to do is change the name of the fn itself (and its test) to `query-param-accept` or something like that to reflect that clarification of its purpose.

**Comment by cap10morgan** [comment]
@danielglauser ping

**Comment by cap10morgan** [comment]
@danielglauser ping ping :)

File: project.clj [hunk 5]
```diff
@@ -1,4 +1,4 @@
-(defproject democracyworks/pedestal-toolbox "0.6.3-SNAPSHOT"
+(defproject democracyworks/pedestal-toolbox "0.6.3"
```
Comment (cap10morgan) [inline project.clj]: PR's should leave `-SNAPSHOT` in the version. We only drop those when we roll a release.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,5 @@
+* 0.6.3 - 2015-10-29
```
Comment (cap10morgan) [inline CHANGELOG.md]: This shouldn't specify a particular version or release date in the PR. I usually just write `HEAD` for the version number to track unreleased changes.

File: src/pedestal_toolbox/params.clj [hunk 7]
```diff
@@ -68,6 +68,22 @@
...
+  browser that sets the Accept header in the normal way.
+
+  To use, add a accept=application/csv to the URL query string.
+  Expects the query params to have already been keywordized by
+  keywordize-params"
+  (interceptor
+   {:enter
+    (fn [ctx]
+      (if-let [content-type (get-in ctx [:request :params :accept])]
+        (assoc-in ctx [:request :headers "accept"] content-type)
```
Comment (cap10morgan) [inline src/pedestal_toolbox/params.clj]: I'm getting nitpicky now, but this is *so* easy to confuse (I've done it a lot in the past), I'd rather see the `content-type` let var renamed `accept-type` or `accept` (or similar).

Comment (danielglauser) [inline src/pedestal_toolbox/params.clj]: Not picky, that's legit

File: test/pedestal_toolbox/params_test.clj [hunk 8]
```diff
@@ -102,3 +102,14 @@
...
                             enter
                             (get-in [:response :body])))))))))
+
+(deftest query-param-accept-test
+  (let [params {"a" 1 :accept "application/csv"}
+        ctx {:request {:params params}}
+        enter (:enter query-param-accept)
+        ctx-with-updated-headers (enter ctx)]
+    (testing "a query param of content type is placed in the headers"
+      (is (= (get-in ctx [:request :params :accept])
```
Comment (cap10morgan) [inline test/pedestal_toolbox/params_test.clj]: I'd rather see a literal comparison value, or a reference to a let'd string of just `"application/csv"` as this test's expected value. Otherwise the more indirection you have in your expected value, the more likely you are to have passing tests where both the expected and actual end up being blank, nil, or some other value you don't want. That seems very unlikely here, but I like to draw bright lines around things. :)

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,5 @@
+* 0.6.3
```
Comment (cap10morgan) [inline CHANGELOG.md]: Almost! Just needs `s/0.6.3/HEAD/`.

**Comment by cap10morgan** [comment]
LGTM
